src/Raven.Server/RavenServer.cs

```diff
@@ -1907,17 +1911,17 @@ private void ListenToNewTcpConnection(TcpListener listener)
                     if (_forTestingPurposes != null && _forTestingPurposes.ThrowExceptionInListenToNewTcpConnection)
                         throw new Exception("Simulated TCP failure.");
 
+                    var tcp = new TcpConnectionOptions
```

Comment:
can this be moved back? there will be leak if sth will fail during AllocateOperationContext or GetMemoryBuffer below

Reply:
Moved